numpy/_core/src/multiarray/alloc.c

```diff
@@ -427,6 +431,10 @@ PyDataMem_SetHandler(PyObject *handler)
     if (handler == NULL) {
         handler = PyDataMem_DefaultHandler;
     }
+    if (!PyCapsule_IsValid(handler, MEM_HANDLER_CAPSULE_NAME)) {
+        PyErr_SetString(PyExc_ValueError, "Capsule must be named 'mem_handler'");
```

Comment:
Does it make sense to throw this exception? This will only run under the hood right?

If it doesn't make sense, what's the appropriate way to handle this?

P.S: This is my first time reading numpy's C code.

Reply:
Yes, I think this is fine, no reason to bike-shed the error type.